### PR TITLE
DF 1782 Update Event Templates

### PR DESCRIPTION
**Pull request by jimmynotjim**
Updated the event templates to match the new data output from the event processors

## Additions

- Added Google Maps image utility macro

## Removals

- Removed right whitespace on single events for tablet/mobile layout

## Changes

- Updated data objects to match processor output
- Updated hero images to center vertically

## Testing

Check out branch and reindex from production content, then navigate to:

- /events/
- /events/fall-credit-union-advisory-council-meeting-in-washington-dc/
- /events/archive/

## Review

- @sebworks 
- @anselmbradford 
- @KimberlyMunoz 
- @ajbush 
- @duelj 
- @schaferjh 

## Preview

screen shot 2015-07-07 at 12 19 50 pm

screen shot 2015-07-07 at 12 20 06 pm

screen shot 2015-07-07 at 12 20 19 pm

[Preview this PR without the whitespace changes](?w=0)

## Notes

- Clicking Tags doesn't correctly filter the events, there's a new task for this for later

### CODE REVIEW

File: _includes/macros/util/format.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+   location_image_url()
+
+   ==========================================================================
+
+   Description:
+
+   Produces a Google Maps image url when given:
+
+   number: A ten-digit string, no spaces.
```
Comment (anselmbradford) [inline _includes/macros/util/format.html]: Says `number` but actual argument is `options`.

Comment (jimmynotjim) [inline _includes/macros/util/format.html]: Sorry, already fixed.

File: _includes/macros/util/format.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+{% macro location_image_url(options={}) %}
+    {%- set location = options.location
+                       | default('Washington, DC', true) -%}
+    {%- set scale = '&scale=' + options.scale | default('2') -%}
+    {%- set size =  '&size='  + options.size  | default('276x155') -%}
+    {%- set zoom =  '&zoom='  + options.zoom  | default('12') -%}
+    {%- set url = 'https://maps.googleapis.com/maps/api/staticmap?center=' -%}
+    {{ url ~ location ~ size ~ zoom ~ scale }}
+{% endmacro %}
```
Comment (anselmbradford) [inline _includes/macros/util/format.html]: Minor: keep EOF newline when merging with flapjack branch.

File: events/_event-landing-macros.html [hunk 3]
```diff
@@ -55,53 +59,62 @@
...
+    {%- set city =  post.venue.address.city + ', ' + post.venue.address.state
+                    | default('Washington, DC') -%}
+    <section class="content_hero hero">
+        <div class="hero_wrapper">
+            <a href="{{ post.permalink }}"
+               class="hero_card hero_card__events">
+                <article class="hero_preview">
+                    {% if post.categories %}
+                    <span class="category-slug">
+                        <!-- TODO replace temp category with post category. -->
```
Comment (anselmbradford) [inline events/_event-landing-macros.html]: MINOR: include in `{# … #}` instead so it doesn't show in the output on the page.

File: events/_event-landing-macros.html [hunk 4]
```diff
@@ -55,53 +59,62 @@
...
+                        {{ category_icon.render('Announcements & updates', 'category-slug_icon') }}
+                        <span class="u-visually-hidden">Category:</span>
+                        {{ 'Event Category Name' }}
```
Comment (anselmbradford) [inline events/_event-landing-macros.html]: Is this text inside jinja as a placeholder?

File: events/_event.html [hunk 5]
```diff
@@ -70,9 +72,10 @@ <h1 class="u-visually-hidden">Event reservation and viewing details</h1>
                           <a href="mailto:reserve@cfpb.gov?subject=Event RSVP&body=To RSVP, please fill in your first and last name below and send this email.%0D%0A%0D%0AFirst name:%0D%0ALast name:" class="btn"><span class="btn_icon__left cf-icon cf-icon-email-social"></span> reserve@cfpb.gov</a>
                       </p>
                     </div>
+                    {% if (event.live_stream) %}
                     <div class="event-status_livestream">
                       <p>
-                          {# TODO Change to livestream icon when that's added to cf-icons. #}
+                          <!-- TODO Change to livestream icon when that's added to cf-icons. -->
```
Comment (anselmbradford) [inline events/_event.html]: Keep as jinja comments.

File: src/static/css/pages/event.less [hunk 6]
```diff
@@ -8,8 +8,10 @@
    ========================================================================== */
 
 .post__event {
-    padding-right: unit(30px / @base-font-size-px, em);
-    margin-right: unit(30px  / @base-font-size-px, em);
+    .respond-to-min(@desktop-min, {
+        padding-right: unit(30px / @base-font-size-px, em);
+        margin-right: unit(30px  / @base-font-size-px, em);
```
Comment (anselmbradford) [inline src/static/css/pages/event.less]: Could the `30px` on L:12 and L:13 be `@grid_gutter-width`?

**Comment by anselmbradford** [comment]
Two minor things:

Add `block__border-right` and `block__border-top` to footer area:

screen shot 2015-07-07 at 2 20 54 pm


Menu items use `&` I'm pretty sure:

screen shot 2015-07-07 at 2 21 30 pm

File: events/_event-landing-macros.html (hunk 4 above)
Comment (jimmynotjim) [inline events/_event-landing-macros.html]: Yes, it was previously existing, not sure why it's showing up as new

File: events/_event.html (hunk 5 above)
Comment (jimmynotjim) [inline events/_event.html]: Whoops, was debugging code and forgot to switch this back

File: events/_event-landing-macros.html [hunk 7]
```diff
@@ -18,25 +18,29 @@
...
-                  {{ event_post_macros.post_summary(post) }}
-              </div>
-          </div>
-      {% else %}
-          <div class="media_body media_body__event">
-            {{ event_post_macros.post_summary(post) }}
-          </div>
-      {% endif %}
-  </article>
+    {%- set city =  post.venue.address.city + ', ' + post.venue.address.state
```
Comment (sebworks) [inline events/_event-landing-macros.html]: Should we be setting the default here or in the macro?

File: events/_event-landing-macros.html (hunk 3 above)
Comment (sebworks) [inline events/_event-landing-macros.html]: Do we need to use a for loop for the event hero?

**Comment by jimmynotjim** [comment]
I think I covered everything posted. Can everyone do another quick run through so we can merge this before tomorrow's meeting? Thanks

**Comment by anselmbradford** [comment]
:+1:

**Comment by sebworks** [comment]
:+1:

**Comment by anselmbradford** [comment]
Sorry to report that the update to `_event-landing-macros.html` has broken the macropolo tests. https://github.com/cfpb/cfgov-refresh/blob/flapjack/_tests/macro_testing/tests/event-landing-macros.json needs update. Can @willbarton take a look?

**Pull request merged**